rust/lance/src/dataset/fragment.rs

```diff
@@ -423,13 +425,66 @@ impl FileFragment {
 
     pub async fn create_from_file(
         filename: &str,
-        schema: &Schema,
+        dataset: &Dataset,
         fragment_id: usize,
         physical_rows: Option<usize>,
     ) -> Result<Fragment> {
-        let fragment =
-            Fragment::with_file_legacy(fragment_id as u64, filename, schema, physical_rows);
-        Ok(fragment)
+        let filepath = dataset.data_dir().child(filename);
+        let file_version =
+            determine_file_version(dataset.object_store.as_ref(), &filepath, None).await?;
```

Comment:
Hmm does that mean someone will need to make `O(num files)` get requests against object store?

Reply:
Yes.  Was there a goal to avoid that?  I noticed the function was async before but not actually doing anything async.